zephyr/Kconfig

```diff
@@ -0,0 +1,44 @@
+# Nordic WiFi driver for nRF52840 and nRF5340
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config WPA_SUPP
+    bool "WPA WPA_SUPP"
+    select EARLY_CONSOLE
+    select NET_SOCKETS
+    select NET_SOCKETS_POSIX_NAMES
```

Comment:
This setting may raise conflicts in case the application is built with POSIX support (CONFIG_POSIX_API). Any reason for not using POSIX in supplicant?

Reply:
You could use `zsock_*` namespace for socket operations to avoid posix dependencies, that's what some Zephyr libraries do.

Reply:
This could be an option in case the dependencies are located only in Zephyr platform specific code but I got an impression that some generic supplicant modules are affected as well (not 100% sure though). Another option as proposed by @rlubos  is letting the application to chose API by having:

`depends on NET_SOCKETS_POSIX_NAMES || POSIX_API`

but then the supplicant would need to support both cases. The general thought is that the supplicant should not forcefully imply POSIX_NAMES nor POSIX_API.

Reply:
In general agree with the suggestion, but the reason for forcing `NET_SOCKETS_POSIX_NAMES` is to avoid changing upstream supplicant code, as it doesn't have os_* wrappers for socket APIs. So, unless we fix that first and upsteam, we need to live with this.

Reply:
Ok, but `POSIX_API` serves a similar purpose as `NET_SOCKETS_POSIX_NAMES` and the former brings a bit more POSIX features, so we can support both flags by removing the `select` and applying the following patch:
```
--- modules/lib/hostap/src/utils/includes.h
+++ modules/lib/hostap/src/utils/includes.h
@@ -44,9 +44,16 @@
 #endif /* CONFIG_NATIVE_WINDOWS */
 
 #if defined(CONFIG_ZEPHYR)
+#if defined(CONFIG_POSIX_API)
+#include <posix/arpa/inet.h>
+#include <posix/sys/select.h>
+#include <posix/sys/socket.h>
+#include <posix/unistd.h>
+#else /* defined(CONFIG_POSIX_API) */
 #include <net/net_ip.h>
 #include <net/socket.h>
+#endif /* defined(CONFIG_POSIX_API) */
 #include <shell/shell.h>
-#endif
+#endif /* defined(CONFIG_ZEPHYR) */
```

We can also add something like:
```
#if !defined(CONFIG_POSIX_API) && !defined(CONFIG_NET_SOCKETS_POSIX_NAMES)
#pragma error "You must enable either ..."
#endif
```
to better inform a user what is missing in their config.

We use `POSIX_API` in Matter project for exactly the same reason - be able to use platform-independent code shared across Zephyr, Linux, Darwin etc, so it's kinda crucial that WPA supplicant build with that flag :)

Reply:
Thanks for the patch. I tried this, but its unable to find `open` definition, so, build fails. `posix/unistd.h` and `lib/os/fdtable.c` also doesn't have a definition for `open`. Am I missing something?

```
libc.a(lib_a-openr.o): in function `_open_r':
openr.c:(.text._open_r+0x14): undefined reference to `_open'
```

Reply:
Its not `fopen` but the `open` system call that is being used and as per https://docs.zephyrproject.org/latest/services/portability/posix.html it is supported.

> Shouldn't we set CONFIG_NO_VLAN for Zephyr or rewrite config_file.c?
Yes, we are taking care of the defined in CMakeLists.txt as they are static and not use configurable.

Reply:
Maybe the documentation is wrong. I can't find a declaration of `open` in Zephyr's `unistd.h` nor elsewhere. Also, I checked with `puncover` tool that the only caller of `open` syscall is `fopen`. Is the tool wrong and you call `open` directly somewhere in `hostap` module?

Reply:
Yes, even I was unable to find a definition. Most code that uses `fopen` in hostapd is unused (except for one call in `src/crypto/random.c`, we directly use `open` for open device files `src/crypto/random.c`, commenting that out using `CONFIG_NO_RANDOM_POOL` builds fine. and we can proceed with that for now, but just wanted to check `open` syscall status in Zephyr.

Reply:
Implemented support for both, and sample uses POSIX as default.